src/main/java/Juan.java

```diff
@@ -0,0 +1,139 @@
+import CustomExceptions.*; // Import custom exception classes
+import TaskChildren.*; // Import task-related classes like ToDo, Deadline, and Event
+
+import java.util.Scanner; // Import Scanner for user input
+
+public class Juan {
+    // Constant for common error message to improve code readability and reusability
+    private final static String porFavor = "Por Favor?\n";
+
+    // Main entry point for the application
+    public static void main(String[] args) {
+        // Display initial line and welcome message
+        lineMessage();
+        helloMessage();
+        lineMessage();
+
+        // Continue chatting as long as user doesn't exit
+        boolean continueChatting = true;
+        while (continueChatting) {
+            // Chat feature to handle user input
+            continueChatting = chatFeature();
+        }
```

Comment:
You can consider extracting this part further to match same level of abstraction in this method.
E.g.
```
readData();
salary = basic * rise + 1000;
tax = (taxable ? salary * 0.07 : 0);
displayResult();
```
versus
```
readData();
processData();
displayResult();
```